akka-distributed-cluster-docs/src/main/paradox/guide/1-event-sourced-shopping-cart.md

```diff
@@ -0,0 +1,132 @@
+The state, commands and events are marked as CborSerializable which is configured to use the built-in CBOR serialization. The template project includes this marker interface CborSerializable:
+
+Scala
+:  @@snip [CborSerializable.scala](/samples/grpc/shopping-cart-service-scala/src/main/scala/shopping/cart/CborSerializable.scala) { }
+
+Java
+:  @@snip [CborSerializable.java](/samples/grpc/shopping-cart-service-java/src/main/java/shopping/cart/CborSerializable.java) { }
+
+Configuration in the application configuration to select the serializer:
+
+Scala
+:  @@snip [application.conf](/samples/grpc/shopping-cart-service-scala/src/main/resources/serialization.conf) { }
+
+Java
+:  @@snip [application.conf](/samples/grpc/shopping-cart-service-java/src/main/resources/serialization.conf) { }
+
+## Client access with Akka gRPC
+
+To allow users to actually use the service we need a public API reachable over the internet. For this we will use @extref[Akka gRPC](akka-grpc:)
+giving us a type safe, efficient protocol that allows clients to be written in many languages.
+
+The service descriptor for the API is defined in protobuf and mirrors the set of commands the entity accepts:
+
+Scala
+:  @@snip [ShoppingCartService.proto](/samples/grpc/shopping-cart-service-scala/src/main/protobuf/ShoppingCartService.proto) { }
+
+Java
+:  @@snip [ShoppingCartService.proto](/samples/grpc/shopping-cart-service-java/src/main/protobuf/ShoppingCartService.proto) { }
+
+When compiling the project the Akka gRPC @scala[sbt]@java[maven] plugin generates a service interface for us to implement.
+Our implementation of it interacts with the entity:
+
+Scala
+:  @@snip [ShoppingCartServiceImpl.scala](/samples/grpc/shopping-cart-service-scala/src/main/scala/shopping/cart/ShoppingCartServiceImpl.scala) {}
+
+Java
+:  @@snip [ShoppingCartServiceImpl.java](/samples/grpc/shopping-cart-service-java/src/main/java/shopping/cart/ShoppingCartServiceImpl.java) {}
+
+Finally, we need to start the HTTP server, making service implementation available for calls from external clients:
+
+FIXME this includes service-to-service publisher, should we create a separate server without that, or just methods perhaps
+
+Scala
+:  @@snip [ShoppingCartServer.scala](/samples/grpc/shopping-cart-service-scala/src/main/scala/shopping/cart/ShoppingCartServer.scala) { #startServer }
+
+Java
+:  @@snip [ShoppingCartServer.java](/samples/grpc/shopping-cart-service-java/src/main/java/shopping/cart/ShoppingCartServer.java) { #startServer }
+
+
+## Running the sample
+
+FIXME we don't really have a separate sample but maybe linking to the grpc-projection one would be fine?
```

Comment:
let's see how ambitious we want to be with the run and download of source code for the steps, or if we just leave that to the end of the tutorial when everything is included

Reply:
Yeah, did something halfways there now, but not sure it is the right thing.